src/main/kotlin/com/intershop/gradle/icm/docker/utils/CustomizationImageBuildPreparer.kt

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2020 Intershop Communications AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.intershop.gradle.icm.docker.utils
+
+import com.intershop.gradle.icm.docker.ICMDockerPlugin.Companion.BUILD_MAIN_IMAGE
+import com.intershop.gradle.icm.docker.ICMDockerPlugin.Companion.BUILD_TEST_IMAGE
+import com.intershop.gradle.icm.docker.tasks.BuildImage
+import com.intershop.gradle.icm.docker.tasks.ProvideResourceFromClasspath
+import org.gradle.api.Project
+import org.gradle.api.tasks.TaskProvider
+import org.gradle.api.tasks.bundling.Tar
+import com.intershop.gradle.icm.docker.extension.Images as BaseImages
+import com.intershop.gradle.icm.docker.extension.image.build.Images as BuildImages
+
+class CustomizationImageBuildPreparer(private val project: Project,
+                                      private val images: BaseImages,
+                                      private val buildImages: BuildImages) {
+
+    companion object {
+        const val NAME_DOCKERFILE = "Dockerfile"
+        const val DIR_DOCKERFILE = "dockerfile"
+        const val RESOURCE_DOCKERFILE = "customization/$NAME_DOCKERFILE"
+        const val ARG_BASE_IMAGE = "BASE_IMAGE"
+        const val ARG_PACKAGE_FILE = "PACKAGE_FILE"
+        const val ARG_NAME = "NAME"
+    }
+
+    fun prepareImageBuilds() {
+
```

Comment:
Why do you do this? There is an task https://bmuschko.github.io/gradle-docker-plugin/api/com/bmuschko/gradle/docker/tasks/image/Dockerfile.html. You can use the name of the package as a parameter for the file creation and it is not necessary to untar the package in a separate step.

Reply:
The task does not work with files that are located inside the classpath (inside a jar file).

Reply:
I think, this is not how Gradle works, because you can not add a second task with the same type. But anyway it's your commit and you will be responsible.

Reply:
There is no second task added:

- provideDockerfile only exists once
- remaining code is adapted from ProjectImageBuildPreparer (lookup and configure tasks createMainPkg and createTestPkg)